src/lib.rs

```diff
@@ -115,6 +115,24 @@ pub mod units {
 
 /// General error-type trait implemented for all error types in this crate
 pub trait Error: fmt::Debug {}
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum TimeError<E: Error = ()> {
+    CastWouldFail,
+    WouldOverflow,
+    WouldUnderflow,
+    WouldDivByZero,
+    DisorderedOperands,
+    Clock(clock::Error<E>),
+}
+impl<E: Error> Error for TimeError<E> {}
+
+impl<E: Error> From<clock::Error<E>> for TimeError<E> {
+    fn from(clock_error: clock::Error<E>) -> Self {
+        TimeError::<E>::Clock(clock_error)
+    }
+}
+
 impl Error for () {}
 impl Error for Infallible {}
```

Comment:
This enum is for any errors that may be generated locally (not implementations). I have it parameterized so it can pass along any impl-specific errors from the `Clock` in the `TimeError::Clock(clock::Error<E>)` variant. Is this the best way to handle passing the impl-specific errors to the user?